demos/free_surface/implicit_top_bottom_buoyancy_free_surface.py

```diff
@@ -0,0 +1,68 @@
+from gadopt import *
+from implicit_top_bottom_free_surface import TopBottomImplicitFreeSurfaceModel
+from test_viscous_surface import run_benchmark
+
+
+class BuoyancyTopBottomImplicitFreeSurfaceModel(TopBottomImplicitFreeSurfaceModel):
+    # Test case from Section 3.1.3 of `An implicit free surface algorithm
+    # for geodynamical simulations', Kramer et al 2012.
+
+    name = "implicit-buoyancy"
+    bottom_free_surface = True
+
+    def __init__(self, dt_factor, nx=320, do_write=False, iterative_2d=False, cartesian=True):
+
+        super().__init__(dt_factor, nx=nx, do_write=do_write, iterative_2d=iterative_2d, cartesian=cartesian)
```

Comment:
I think you're only overriding the `nx` default here:

```suggestion
    def __init__(self, dt_factor, nx=320, **kwargs):
        super().__init__(dt_factor, nx=nx, **kwargs)
```

Reply:
sounds good